rust/lance-linalg/src/distance/l2.rs

```diff
@@ -166,14 +166,27 @@ mod x86_64 {
                 debug_assert_eq!(from.len(), to.len());
 
                 // Get the potion of the vector that is aligned to 32 bytes.
-                let len = from.len() / 8 * 8;
+                let len = from.len() / 32 * 32;
                 let mut sums = _mm256_setzero_ps();
-                for i in (0..len).step_by(8) {
+                for i in (0..len).step_by(32) {
                     let left = _mm256_loadu_ps(from.as_ptr().add(i));
+                    let l2 = _mm256_loadu_ps(from.as_ptr().add(i + 8));
+                    let l3 = _mm256_loadu_ps(from.as_ptr().add(i + 16));
+                    let l4 = _mm256_loadu_ps(from.as_ptr().add(i + 24));
+
                     let right = _mm256_loadu_ps(to.as_ptr().add(i));
+                    let r2 = _mm256_loadu_ps(to.as_ptr().add(i + 8));
+                    let r3 = _mm256_loadu_ps(to.as_ptr().add(i + 16));
+                    let r4 = _mm256_loadu_ps(to.as_ptr().add(i + 24));
                     let sub = _mm256_sub_ps(left, right);
                     // sum = sub * sub + sum
+                    let s2 = _mm256_sub_ps(l2, r2);
+                    let s3 = _mm256_sub_ps(l3, r3);
+                    let s4 = _mm256_sub_ps(l4, r4);
                     sums = _mm256_fmadd_ps(sub, sub, sums);
+                    sums = _mm256_fmadd_ps(s2, s2, sums);
+                    sums = _mm256_fmadd_ps(s3, s3, sums);
```

Comment:
you can use 4 accumulators for this `sums1..4` and only add the four together at the end. Might be better throughput

Reply:
Done.